### PR TITLE
core(session): ignore protocol timeouts from puppeteer

### PR DESCRIPTION
**Pull request by adamraine**
Fixes https://github.com/GoogleChrome/lighthouse/issues/15510

### CODE REVIEW

File: core/gather/session.js [hunk 1]
```diff
@@ -93,7 +93,13 @@ class ProtocolSession extends CrdpEventEmitter {
       }));
     });
 
-    const resultPromise = this._cdpSession.send(method, ...params);
+    const resultPromise = this._cdpSession.send(method, ...params).catch(err => {
+      // We set up our own protocol timeout system, so we should ignore protocol timeouts emitted by puppeteer.
+      // https://github.com/GoogleChrome/lighthouse/issues/15510
+      if (/'protocolTimeout'/.test(err)) return;
```
[the remark this answers is not in the document]
Comment (adamraine) [reply core/gather/session.js:99]: It is, I think it get's sent through `toString()` for regex `.test()`

File: core/gather/session.js [hunk 2]
```diff
@@ -94,7 +94,13 @@ class ProtocolSession extends CrdpEventEmitter {
...
```
Comment (brendankenny) [inline core/gather/session.js:100]: Won't this resolve with `undefined`, though, if `timeoutPromise` doesn't win the race?

Comment (adamraine) [reply core/gather/session.js:100]: Good point. I guess if puppeteer gives us a protocol timeout error that means they are done processing the command. Even if we wait around longer, we won't get a result.

Comment (adamraine) [reply core/gather/session.js:100]: I suppose an better solution would be to just ignore this type of error for our `Page.navigate` call specifically? IDK but this clearly needs more work.

**Comment by adamraine** [comment]
This won't work because puppeteer will cut off the command response even if Lighthouse is still waiting. I think what we need is for puppeteer to allow setting a timeout on individual commands.

**Pull request closed without merging**